ui/app/src/components/Search/Search.tsx

```diff
@@ -70,6 +74,23 @@ export function Search(props: SearchProps) {
     searchParameters,
     onSelect
   });
+
+  useMount(() => {
+    // Set initial filters coming from props to checkedFilters
+    if (parameters?.filters) {
+      const newCheckedFilters = {};
+      Object.entries(parameters.filters).forEach(([key, filter]: [string, any]) => {
+        if (Array.isArray(filter)) {
+          newCheckedFilters[key] = filter.reduce((checked, key) => ({ ...checked, [key]: true }), {});
+        } else if (filter.date) {
+          newCheckedFilters[key] = `${filter.min ?? ''}TODATE${filter.max ?? ''}ID${filter.id}`;
+        } else {
+          newCheckedFilters[key] = `${filter.min ?? ''}TO${filter.max ?? ''}`;
+        }
+      });
```

Comment:
Preferably this should be in a single place. Where does this concatenation and such usually occurs? I would have thought this all should be on the useSearchState hook.